### PR TITLE
feat: Add C code for the section of intro to DP

### PR DESCRIPTION
**Pull request by huawuque404**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [√] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [√] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [√] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Review by Gonglja — COMMENTED** [review]
Hi, you don’t need to close the last PR. 
You can just modify and submit it directly after putting forward your comments. 
Otherwise, you won’t be able to see your previous comments.

Association #766

File: codes/c/chapter_dynamic_programming/climbing_stairs_backtrack.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "../utils/common.h"
+
+/* 回溯 */
+void backtrack(int *choices, int state, int n, int *res) {
+    // 当爬到第 n 阶时，方案数量加 1
+    if (state == n)
+        res[0]++;
+    // 遍历所有选择
+    for (int i = 0; i < sizeof(choices) / sizeof(int); i++) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/climbing_stairs_backtrack.c:15]: This length calculation is not accurate. After passing the pointer into the function through the parameter, the array has degenerated into a pointer. At this time, it is wrong to calculate it through the sizeof form.
图片

**Review by Gonglja — COMMENTED** [review]
**Please note, do not reopen PR again.
Just modify the local code directly, then commit and push.**

File: codes/c/chapter_dynamic_programming/climbing_stairs_backtrack.c (hunk 1 above)
Comment (huawuque404) [reply codes/c/chapter_dynamic_programming/climbing_stairs_backtrack.c:15]: Ok this is a very low-level mistake! Thank you for your correction

Comment (huawuque404) [reply codes/c/chapter_dynamic_programming/climbing_stairs_backtrack.c]: It has been modified.

**Review by Gonglja — COMMENTED** [review]
Approve, @krahets

**Review by krahets — APPROVED** [review]

**Pull request merged**